test/parallel_api/experimental/asynch.pass.cpp

```diff
@@ -207,7 +218,8 @@ test_with_usm()
     // call third algorithm that has to wait for first to complete
     auto new_policy3 = TestUtils::make_device_policy<
         TestUtils::unique_kernel_name<Async<3>, TestUtils::uniq_kernel_index<alloc_type>()>>(q);
-    oneapi::dpl::experimental::sort_async(new_policy3, data2, data2 + n, fut1);
+    auto sort_async_result = oneapi::dpl::experimental::sort_async(new_policy3, data2, data2 + n, fut1);
```

Comment:
I think the root issue was here: we may receive `__future` object from this call with temporary data and breaks their keep-a-live here here.